x/stakeibc/keeper/msg_server.go

```diff
@@ -1103,3 +1106,66 @@ func (k msgServer) ResumeHostZone(goCtx context.Context, msg *types.MsgResumeHos
 
 	return &types.MsgResumeHostZoneResponse{}, nil
 }
+
+// Submits an ICA tx to either grant or revoke authz permisssions to an address
+// to execute trades on behalf of the trade ICA
+func (k msgServer) ToggleTradeController(
+	goCtx context.Context,
+	msg *types.MsgToggleTradeController,
+) (*types.MsgToggleTradeControllerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Fetch the trade ICA which will be the granter
+	tradeRoute, found := k.GetTradeRouteFromTradeChainId(ctx, msg.ChainId)
+	if !found {
+		return nil, types.ErrTradeRouteNotFound.Wrapf("trade route not found for chain ID %s", msg.ChainId)
+	}
+
+	swapMsgTypeUrl := "/" + proto.MessageName(&types.SwapAmountInRoute{})
+
+	// Build the authz message, depending on whether permissions are granted or revoked
+	// The granter will always be the trade ICA and the grantee will be the address
+	// specified in the message
+	var authzMsg []proto.Message
+	switch msg.PermissionChange {
+	case types.AuthzPermissionChange_GRANT:
+		authzMsg = []proto.Message{&authz.MsgGrant{
+			Granter: tradeRoute.TradeAccount.Address,
+			Grantee: msg.Address,
+			Grant:   authz.Grant{
```

Comment:
Was this supposed to be the msgTypeUrl which you made?

Reply:
It takes an `Authorization` object - which has the type url as a component. Just pushed the update